src/inventory_ui.h

```diff
@@ -48,18 +49,7 @@ struct inventory_input;
 using drop_location = std::pair<item_location, int>;
 using drop_locations = std::list<drop_location>;
 
-struct inventory_entry_drawn_info {
-    int text_x_start;
-    int text_x_end;
-    int y;
-    bool include_point( point p )const {
-        if( text_x_start <= p.x &&
-            p.x <= text_x_end &&
-            y == p.y ) {
-            return true;
-        }
-        return false;
-    }
+struct inventory_entry_drawn_info : entry_drawn_info {
```

Comment:
Why not use `entry_drawn_info` directly? This inheritance does not add anything to the parent class.

Reply:
in future there may be something added

Reply:
You can always create a new class in the future, but now it does nothing and only clutters up the code. And even if something is to be added in the future, I would recommend adding it to the parent class (any maybe guard it with a flag), so other UIs can also use the new functionality when they need to.

Reply:
I mean there are a lot of variant UI things in game, else than inventory and uilist, there are pickup, inventory_item_menu, crafting, and whatever, each of them are having different kind of drawing, its just this two drawn info structs are currently not different.

Reply:
besides, I am also considering to add some new things, such as button.

Reply:
> I mean there are a lot of variant UI things in game, else than inventory and uilist, there are pickup, inventory_item_menu, crafting, and whatever, each of them are having different kind of drawing

And I'm pretty sure it's a bad thing we'd like to get rid of and have a unified system to display menu stuff instead.